modules/friendlytag.js

```diff
@@ -1755,7 +1755,7 @@ Twinkle.tag.callbacks = {
 					oldPageTags += '\n' + pageTag;
 				});
 			}
-			pageText += '\n{{Redirect category shell|' + tagText + oldPageTags + '\n}}';
+			pageText += '\n\n{{Redirect category shell|' + tagText + oldPageTags + '\n}}';
```

Comment:
```suggestion
			pageText = pageText.trim() + '\n\n{{Redirect category shell|' + tagText + oldPageTags + '\n}}';
```
not sure if it's possible for pageText to already have some newlines, if it does this change would fix it